zeq/connection/broker.h

```diff
@@ -0,0 +1,56 @@
+
+/* Copyright (c) 2014-2015, Human Brain Project
+ *                          Stefan.Eilemann@epfl.ch
+ */
+
+#ifndef ZEQ_CONNECTION_BROKER_H
+#define ZEQ_CONNECTION_BROKER_H
+
+#include <zeq/receiver.h> // base class
+#include <lunchbox/debug.h>
+
+namespace zeq
+{
+namespace connection
+{
+namespace detail { class Broker; }
+
+/**
+ * Brokers subscription requests for a zeq::Receiver.
+ *
+ * Example: @include tests/connection/broker.cpp
+ */
+class Broker : public Receiver
+{
+public:
+    /**
+     * Create a new subscription broker.
+     *
+     * The given receiver has to have at least the same lifetime as this
+     * broker. The receiver and broker are automatically shared.
+     *
+     * For simplicity, only a single Receiver is handled by a Broker. The
+     * implementation should be extended if multiple receivers shall be
+     * handled.
+     *
+     * @param address the zmq reply socket address to be used.
+     * @param receiver the Receiver to manage.
+     * @throw std::runtime_error when the zmq setup failed.
+     */
+    ZEQ_API Broker( const std::string& address, Receiver& receiver );
```

Comment:
Sorry, my fault. In my mindset it's still (1 socket <=> 1 message type/semantic message channel), which entails a number of receivers per component, which in turn means handling a number of brokers. Hard to get rid of old habits, even over the turn of the year...